doc: better security rules documentation (#1305) (#1308)
micronaut-projects/micronaut-security: src/main/docs/guide/securityRule/interceptUrlMap.adoc

- accessing `/v1/myResource/**` with a GET request does not require authentication
- accessing `/v1/myResource/**` with a request that isn't GET requires authentication

> IMPORTANT: When a ann:security.annotation.Secured[] annotation is provided with a set of roles, then a user will have to access to that resource if they have *at least one* of thoose roles, not all.

Reviewer: ```suggestion
IMPORTANT: When the ann:security.annotation.Secured[] annotation has a set of roles, the <<secured, SecuredAnnotationRule>> grants access to a user if they have **any** of the roles.
```
